pkg/api/v1/status/condition.go

```diff
@@ -66,7 +66,22 @@ func TrueCondition(conditionType ConditionType) Condition {
 		Type:               conditionType,
 		Status:             corev1.ConditionTrue,
 		LastTransitionTime: metav1.Now(),
-		Reason:             "",
-		Message:            "",
 	}
 }
+
+// FalseCondition returns the COndition that has the 'Status' set to 'false' and 'Type' to 'conditionType'.
+// The reason and message can be provided optionally
+func FalseCondition(conditionType ConditionType, reasonAndMessage ...string) Condition {
```

Comment:
[nit]
That looks OK for tests, but in actual code we may want to preserve the `workflow.Reason` type (i.e. by using `conditions.New(type, false).WithReason(reason)` or something of that sort)

Reply:
I think you are right, I was too lazy and decided to postpone `with` solution for later.
I'll address this in the next PR which is dependent on this one and is on-going.
thanks 👍